src/libraries/Microsoft.Extensions.Configuration.Binder/tests/ConfigurationBinderTests.cs

```diff
@@ -812,6 +817,39 @@ public void CanBindValueTypeOptions()
             Assert.Equal("hello world", options.MyString);
         }
 
+        [Fact]
+        public void CanBindByteArray()
+        {
+            var bytes = new byte[] { 1, 2, 3, 4 };
+            var dic = new Dictionary<string, string>
+            {
+                { "MyByteArray", Convert.ToBase64String(bytes) }
+            };
+            var configurationBuilder = new ConfigurationBuilder();
+            configurationBuilder.AddInMemoryCollection(dic);
+            var config = configurationBuilder.Build();
+
+            var options = config.Get<ByteArrayOptions>();
+            Assert.Equal(bytes, options.MyByteArray);
+        }
+
+        public void ExceptionWhenTryingToBindToByteArray()
+        {
+            var dic = new Dictionary<string, string>
+            {
+                { "MyByteArray", "(not a valid base64 string)" }
+            };
+            var configurationBuilder = new ConfigurationBuilder();
+            configurationBuilder.AddInMemoryCollection(dic);
+            var config = configurationBuilder.Build();
+
+            var exception = Assert.Throws<InvalidOperationException>(
+                () => config.Get<ByteArrayOptions>());
+            Assert.Equal(
```

Comment:
We usually don't assert exception messages as they could be localized.

Reply:
I followed the example of other tests in the same class, and they all assert the message. Should I do differently here?
Localization shouldn't be an issue, since `SR.Error_FailedBinding` follows the current culture anyway.

Reply:
@safern so, should I remove this assertion, or leave it to be consistent with the other tests?

Reply:
I think it's fine to keep it since we're already doing it in other tests from this test library.